backend/endpoints/rooms.py

```diff
@@ -353,3 +353,92 @@ async def close_conversation(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="unable to close conversation",
     )
+
+
+@router.get(
+    "/org/{org_id}/rooms/{room_id}/members",
+    response_model=ResponseModel,
+    status_code=status.HTTP_200_OK,
+    responses={
+        400: {"detail": "Invalid keyword supplied"},
+        404: {"detail": "Room not found"},
+        424: {"detail": "Failure to retrieve room members"},
+    },
+)
+async def get_members(org_id: str, room_id: str, keyword: Optional[str] = None):
+
+    """Get room members.
+    Returns all members in a room if the room is found in the database
+    Returns filtered members in a room if keyword is supplied
+    Raises HTTP_400_BAD_REQUEST if keyword supplied is not admin or member
+    Raises HTTP_404_NOT_FOUND if the room is not found
+    Raises HTTP_424_FAILED_DEPENDENCY if there is an error retrieving the room members
+    Args:
+        org_id (str): A unique identifier of an organisation
+        room_id (str): A unique identifier of the room
+    Returns:
+        HTTP_200_OK (Room members retrieved successfully):
+
+        {
+            "status": "success",
+            "message": "Room members retrieved",
+            "data": {
+                "61696f5ac4133ddaa309dcfe": {
+                "closed": false,
+                "role": "admin",
+                "starred": false
+                },
+                "6169704bc4133ddaa309dd07": {
+                "closed": false,
+                "role": "admin",
+                "starred": false
+                }
+            }
+        }
+
+    Raises:
+        HTTPException [400]: Invalid keyword supplied
+        HTTPException [404]: Room not found
+        HTTPException [424]: Failure to retrieve room members
+    """
+    room = await get_room(org_id, room_id)
+
+    if not room:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Room doesn't exist"
+        )
+
+    members = room["room_members"]
+    if members and members.get("status_code") is not None:
+        raise HTTPException(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY,
+            detail="Failure to retrieve room members",
+        )
+
+    if keyword:  # if keyword is supplied
+        if keyword not in ["admin", "member"]:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Invalid keyword supplied",
```

Comment:
So the detail could be "invalid role"

Reply:
I created the code with the extra filters initially and I still have it in my codebase. 

So I've decided that since the consensus is that it might not be necessary but we would add it later, I'll just push that instead. 

Since majority have an issue with the name of the filter when it checked for just the role. 

With the new code that has the check for the extra "keywords", what do you suggest I rename it to? 

@zxenonx @o4codes @man-of-mind